makefiles/features_check.inc.mk

```diff
@@ -1,20 +1,28 @@
 # Check if all required FEATURES are provided
 
+# Features that are used without taking "one out of" dependencies into account
+FEATURES_USED_SO_FAR := $(sort $(FEATURES_REQUIRED) $(FEATURES_OPTIONAL_USED))
+
+_features_would_conflict := $(if $(findstring $1,$2),$(filter-out $1,$(subst :, ,$2)))
+FEATURES_WOULD_CONFLICT := $(foreach features,$(FEATURES_USED_SO_FAR),\
+                             $(call _features_would_conflict,$(features),$(FEATURES_CONFLICT)))
```

Comment:
What is the difference between this and the `_features_conflicting` and `_features_used_conflicting` functions?

Reply:
These check for potential conflicts in order to avoid them. The existing functions check for actual conflicts that did happen.

The reason why cannot be used for the other is that checking for potential conflicts means for each set of incompatible features you want to filter only those containing a feature that is used and remove the already used features from them. Those are now treated as pseudo-blacklisted, as using them would result in a conflict.

For the other cases one is interested for sets that contain more than one used feature, and you want to print the intersection of the conflicting feature in one set and the actually used features. I just see no way one can be expressed reusing the other. (But that also might be due to either my lack of Makefile-fu or the lack of expressiveness of Makefiles, not by concept.)

Reply:
Ok, no worries, but mind adding some comments to make those functions a bit more understandable on first sight?

Reply:
Oh, there is a logic error in the code. I have to rewrite the function :-(

Reply:
Fixed and documented